indico/modules/events/registration/badges.py

```diff
@@ -209,3 +200,263 @@ def _build_pdf(self, canvas):
+        """Create/patch an object.
+
+        Args:
+            issuer_id (str): The issuer ID being used for this request.
+            class_suffix (str): Developer-defined unique ID for the pass class.
+            object_suffix (str): Developer-defined unique ID for the pass object.
+            update (bool): Update existing object template.
+
+        Returns:
+            msg (str): The response json format object.
+        """
+
+        # Check if the object exists
+        response = self.http_client.get(
+            url=f'{self.object_url}/{issuer_id}.{object_suffix}')
+
+        if response.status_code == 200:
+            if update:
+                updated_object = self.create_object_template(issuer_id, class_suffix, object_suffix)
+                response = self.http_client.put(url=f'{self.object_url}/{issuer_id}.{object_suffix}',
+                                                json=updated_object)
+                if response.status_code != 200:
+                    raise BadRequest(response.text)  # Something else went wrong...
+            return response.json()
+        elif response.status_code != 404:
+            raise BadRequest(response.text)  # Something else went wrong...
+
+        # See link below for more information on required properties
+        # https://developers.google.com/wallet/tickets/events/rest/v1/eventticketobject
+        new_object = self.create_object_template(issuer_id, class_suffix, object_suffix)
+        # Create the object
+        response = self.http_client.post(url=self.object_url, json=new_object)
+        return response.json()
+
+    def get_link(self, ticket_class: str, ticket_object: str) -> str:
+        """Create a Google Wallet link.
+
+        Args:
+            ticket_class (str): The Ticket class json.
+            ticket_object (str): The Ticket Object json.
+
+        Returns:
+            link (str): The full Google Wallet link.
+        """
+        # Create the JWT claims
+        try:
+            if not (self.registration or self.event):
+                raise ValueError('Missing Registration or Event')
+            claims = {
+                'iss': self.credentials.service_account_email,
+                'aud': 'google',
+                'origins': [config.BASE_URL],
+                'typ': 'savetowallet',
+                'payload': {
+                    # The listed classes and objects will be created
+                    'eventTicketClasses': [ticket_class],
+                    'eventTicketObjects': [ticket_object]
+                }
+            }
+
+            # The service account credentials are used to sign the JWT
+            signer = crypt.RSASigner.from_service_account_info(self.settings['google_wallet_application_credentials'])
+            token = jwt.encode(signer, claims).decode('utf-8')
+        except Exception as e:
+            Logger.get('events.registration').warning('Cannot create Google Wallet link: %s', str(e))
```

Comment:
And the request just fails in that case, because `token` may be undefined below. In any case , returning something is pointless when you have no token.

I'd rather not have any exception handling here and let it fail nicely with the default exception handling Indico already provides...

Reply:
OK, but I'll log a WARNING and not an ERROR, to avoid being flooded by Emails.

Reply:
We do not need custom logging for this. People can already use sentry and/or disable emails - and any exception will go there anyway. 

Anyway, my point was: If this is something that is not expected to fail, then do not handle the exception and let it go through the standard exception handling (which triggers an email, logs to sentry, or whatever). If this is something that's expected to happen, e.g. because Google's server are sometimes unstable, then the exception should be caught and a more friendly "try again later" or similar error shown to the user. In *that* case logging it as a warning makes perfect sense, because it's something you handled, and you just want it in the log as well.

However, in this *particular* block of code I don't see anything that touches the network, so I don't see how this would occasionally fail. Hence, no error handling should be done here IMHO.